src/ts/components/ecomodecontainer.ts

```diff
@@ -0,0 +1,107 @@
+import { PlayerAPI, SegmentPlaybackEvent } from 'bitmovin-player';
+import { i18n } from '../localization/i18n';
+import { Container, ContainerConfig } from './container';
+import { EcoModeToggleButton } from './ecomodetogglebutton';
+import { Label, LabelConfig } from './label';
+import { SettingsPanelItem } from './settingspanelitem';
+
+export class EcoModeContainer extends Container<ContainerConfig> {
+  private ecoModeSavedEnergyItem: SettingsPanelItem;
+  private energySavedLabel: Label<LabelConfig>;
+  private savedEnergy = 0;
+  private currentEnergyEmission: number;
+  private maxEnergyEmisson: number;
+
+  constructor(config: ContainerConfig = {}) {
+    super(config);
+
+    const ecoModeToggleButton = new EcoModeToggleButton();
+    const labelEcoMode = new Label({
+      text: i18n.getLocalizer('ecoMode.title'),
+      for: ecoModeToggleButton.getConfig().id,
+      id: 'ecoModeLabel',
+    } as LabelConfig);
+    this.energySavedLabel = new Label({
+      text: '',
+      cssClass: 'ui-label-savedEnergy',
+    } as LabelConfig);
+
+    const ecoModeToggleButtonItem = new SettingsPanelItem(labelEcoMode, ecoModeToggleButton);
+    this.ecoModeSavedEnergyItem = new SettingsPanelItem('Saved Energy', this.energySavedLabel, { hidden: true });
+
+    this.addComponent(ecoModeToggleButtonItem);
+    this.addComponent(this.ecoModeSavedEnergyItem);
+
+    ecoModeToggleButton.onToggleOn.subscribe(() => {
+      this.ecoModeSavedEnergyItem.show();
+      this.onActivecallback();
+    });
+
+    ecoModeToggleButton.onToggleOff.subscribe(() => {
+      this.ecoModeSavedEnergyItem.hide();
+      this.onActivecallback();
+    });
+  }
+
+  private onActivecallback: () => void;
+
+  public setOnActiveChangeCallback(callback: () => void) {
+    this.onActivecallback = callback;
+  }
+
+  configure(player: PlayerAPI): void {
+    player.on(player.exports.PlayerEvent.SegmentPlayback, (segment: SegmentPlaybackEvent) => {
+      const { height, width, bitrate, frameRate } = segment.mediaInfo;
+
+      const maxQualityAvailable = player.getAvailableVideoQualities().length - 1;
+      const maxHeight = player.getAvailableVideoQualities()[maxQualityAvailable].height;
+      const maxBitrate = player.getAvailableVideoQualities()[maxQualityAvailable].bitrate;
+      const maxWidth = player.getAvailableVideoQualities()[maxQualityAvailable].width;
+
+      const currentEnergyKwh = this.calculateEnergyConsumption(frameRate, height, width, bitrate);
+
+      const maxEnergyKwh = this.calculateEnergyConsumption(frameRate, maxHeight, maxWidth, maxBitrate);
+
+      if (this.ecoModeSavedEnergyItem.isActive()) {
+        this.energySaved(currentEnergyKwh, maxEnergyKwh, this.energySavedLabel);

```

Comment:
We should discuss this. I _think_ instead of using the energy consumption in kWh we might need to take the energy consumption per second and multiply it by the segment duration?